frame/assets/src/lib.rs

```diff
@@ -163,35 +282,158 @@ pub mod pallet {
 		/// The origin which may forcibly create or destroy an asset.
 		type ForceOrigin: EnsureOrigin<Self::Origin>;
 
-		/// The basic amount of funds that must be reserved when creating a new asset class.
-		type AssetDepositBase: Get<BalanceOf<Self>>;
-
-		/// The additional funds that must be reserved for every zombie account that an asset class
-		/// supports.
-		type AssetDepositPerZombie: Get<BalanceOf<Self>>;
-
-		/// The maximum length of a name or symbol stored on-chain.
-		type StringLimit: Get<u32>;
+		/// The basic amount of funds that must be reserved for an asset.
+		type AssetDeposit: Get<DepositBalanceOf<Self>>;
 
 		/// The basic amount of funds that must be reserved when adding metadata to your asset.
-		type MetadataDepositBase: Get<BalanceOf<Self>>;
+		type MetadataDepositBase: Get<DepositBalanceOf<Self>>;
 
 		/// The additional funds that must be reserved for the number of bytes you store in your
 		/// metadata.
-		type MetadataDepositPerByte: Get<BalanceOf<Self>>;
+		type MetadataDepositPerByte: Get<DepositBalanceOf<Self>>;
+
+		/// The amount of funds that must be reserved when creating a new approval.
+		type ApprovalDeposit: Get<DepositBalanceOf<Self>>;
+
+		/// The maximum length of a name or symbol stored on-chain.
+		type StringLimit: Get<u32>;
 
 		/// Weight information for extrinsics in this pallet.
 		type WeightInfo: WeightInfo;
 	}
 
+	#[pallet::storage]
+	/// Details of an asset.
+	pub(super) type Asset<T: Config> = StorageMap<
+		_,
+		Blake2_128Concat,
+		T::AssetId,
+		AssetDetails<T::Balance, T::AccountId, DepositBalanceOf<T>>,
+	>;
+
+	#[pallet::storage]
+	/// The number of units of assets held by any given account.
+	pub(super) type Account<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		T::AssetId,
+		Blake2_128Concat,
+		T::AccountId,
+		AssetBalance<T::Balance>,
+		ValueQuery,
+	>;
+
+	#[pallet::storage]
+	/// Approved balance transfers. First balance is the amount approved for transfer. Second
+	/// is the amount of `T::Currency` reserved for storing this.
+	pub(super) type Approvals<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		T::AssetId,
+		Blake2_128Concat,
+		ApprovalKey<T::AccountId>,
```

Comment:
Should we invent and use a `StorageTripleMap`?

Reply:
would make indexing through approvals a bit easier if that was something we wanted to do.

Reply:
another PR, perhaps? cc @thiolliere

Reply:
I opened an issue for tracking: https://github.com/paritytech/substrate/issues/8291